File: JavaScript/src/bin/easy_73.js
/*
 * Daily Coding Problem: 73
 * Difficulty: Easy
 * Asked By: Google
 * Authored By: TenthEdict
 */

class Node {
    constructor(val, next=null) {
        this.val = val;
        this.next = next;
    }
}

function reverse(head) {
    let prev = null;
    let curr = head;

    while (curr) {
        let next = curr.next;
        curr.next = prev;
        prev = curr;
        curr = next;
    }
    return prev;
}

function print_list(head) {
    while (head) {
        console.log(head.val);
        head = head.next
    }
}

let node3 = new Node(3);
let node2 = new Node(2, node3);
let node1 = new Node(1, node2);

console.log(print_list(reverse(node1)))

File: Python/src/bin/easy_73.py
# Daily Coding Problem: 73
# Difficulty: Easy
# Asked by: Google
# Authored by: TenthEdict

class Node:
    def __init__(self, val, next=None):
        self.val = val
        self.next = next

def reverse(head):
    prev = None
    curr = head

    while curr:
        next = curr.next
        curr.next = prev
        prev = curr
        curr = next

    return prev

def print_list(head):
    while(head):
        print(head.val)
        head = head.next


def main():
    node3 = Node(3)
    node2 = Node(2, node3)
    node1 = Node(1,node2)

    print_list(reverse(node1))

main()
